main.rb

```diff
@@ -0,0 +1,29 @@
+require_relative 'codebraker'
+require_relative 'cli'
+
+class Main
+  attr_accessor :current_attempts
+
+  def initialize(current_attempts = 0)
+    @codebraker = Codebraker.new
+    @current_attempts = current_attempts
+  end
+
+  def run
+    CLI.greeting
+    @current_attempts = CLI.choose_attempts_number
+    @current_attempts.times do
+      entered_number = Codebraker.enter_number
+      hidden_number = @codebraker.check(entered_number)
+      if CLI.won_game?(hidden_number)
+        break
+      else
+        CLI.game_over(current_attempts = @current_attempts)
```

Comment:
It's senseless method why do you check if game_over in cycle. If cycle finished - it means that game was over

Reply:
Fixed